docs/proto-docs.md

```diff
@@ -1172,6 +1190,205 @@ Msg defines the attribute module Msg service.
 
 
 
+<a name="provenance/hold/v1/events.proto"></a>
+<p align="right"><a href="#top">Top</a></p>
+
+## provenance/hold/v1/events.proto
+
+
+
+<a name="provenance.hold.v1.EventHoldAdded"></a>
+
+### EventHoldAdded
+EventHoldAdded is an event indicating that some funds were placed on hold in an account.
+
+
+| Field | Type | Label | Description |
+| ----- | ---- | ----- | ----------- |
+| `address` | [string](#string) |  | address is the bech32 address string of the account with the funds. |
+| `amount` | [string](#string) |  | amount is a Coins string of the funds placed on hold. |
```

Comment:
we should consider adding a free form "reason" string to this event

Reply:
Good call.

I added it to the `EventHoldAdded`. I added a `reason` argument to the `AddHold` keeper function which is passed directly into the event.

I didn't put any restrictions on it though. It can be empty or ginormous.

Arguments for no limits:
1. I expect the value to always be code generated. As such, an error for it would be really annoying from a user's standpoint.
2. Along those same lines, I didn't really want to add a new failure point to `AddHold`.
3. The max would be completely arbitrary anyway. The reason for a max is to save space, but there isn't a specific number from a technical standpoint. My first thought was 100 charcters, but I'm sure there will end up being legitimate uses that might want more than that. So then I thought about 250 or 1000. But I realized that I was trying to come up with a number that would never actually cause an error. If I never wanted it to cause an error, the answer is to not have a max.
4. I decided to allow an empty string due to the same reasons as not having a max. I.e. "What's a good minimum length." A one character reason is only barely less worthless as an empty reason; it's supposed to be human readable.
5. It'd be easy to add a limit later if some problematic reasons are being used. At that time, we'll know more about what not to allow.

Reply:
I was thinking this is similar to the reason given on gas meter usage ... something that essentially is an event from the point of use and is setup by code ... I don't see any reason to really put much structure around it, empty or otherwise it would be up to the caller to put something that makes sense in there.